### DISCUSSION TITLE
Custom Input Format

### DISCUSSION THREAD

**Discussion by ddaniels** [ClickHouse/ClickHouse#63981] [Q&A] [answered]
Hi,

Is it possible to write a [custom input format](https://clickhouse.com/docs/en/interfaces/formats)? We have a binary format with custom headers that I'd like to be able to read into clickhouse. It's somewhat similar to the RowBinary format, but has some differences as well.

Thanks,
Doug

**Comment by yiyang-shao** [comment, marked as answer]
Hi Doug!

The custom input format is not exposed as a feature. But you may make some modification in ClickHouse code. Take `src/Processors/Formats/Impl/BinaryRowInputFormat.cpp` as a reference.
